src/image_preprocess.py

```diff
@@ -117,22 +156,74 @@ def normalize_img(img, scaling='minmax'):
         # (X - min) / (max - min)
         return np.interp(img, (img.min(), img.max()), (0, +1))
     elif scaling == 'zscore':
-        flat_img = img.flatten()
-        pixel_mean = np.mean(flat_img)  # get mean
-        pixel_std = np.std(flat_img)  # get standard deviation
-        return (img - pixel_mean) / pixel_std  # (X - mean) / std
+        return (img - img.mean()) / img.std()  # (X - mean) / std
     else:
-        raise NotImplementedError('Currently only minmax normalization and \
-                                  zscore standardization are supported.')
+        raise TypeError("'{}' invalid keyword argument. ".format(scaling) +
+                        "Only minmax normalization and "
+                        "zscore standardization are supported.")
 
 
-def get_dataset(configs):
-    """
-    configs["dataset_root"] = "path/to/root/of/datset"
-    --- use this to grab images, do whatever preprocessing is needed,
-    and output like this:
+def get_dataset(configs, height, width, scaling='minmax'):
+    """Grabs images, preprocesses them, and returns list of tensors.
+    Also preprocesses annotations.
+
+    Arguments:
+        configs {dict} -- Contains config options for training.
+                          Uses "dataset_root" and "batch_size".
+                                 should contain both images and annotation files.
+        height {int} -- Pixel height required by the model.
+        width {int} -- Pixel width required by the model.
+
+    Keyword Arguments:
+        scaling {str} -- Describes the normalization method. (default: {'minmax'})
 
     Returns:
-        List: tensor images (3x300x300), list: annotations
+        List: tensor batches (Bx300x300x3), List: List: tensor (Nx4)
+        -- Two ordered lists (image batches, annotation batch lists). The
+           annotation batch lists contain tensors of varying length due to
+           there being different number of annotations per image.
     """
-    raise NotImplementedError
+    dataset_path = configs["dataset_root"]
+    batch_size = configs["batch_size"]
+    # files, sorted so that img & annots are in same order
+    image_names = sorted([i for i in listdir(dataset_path) if i[-4:] == '.png'])
+    ann_names = sorted([i for i in listdir(dataset_path) if i[-5:] == '.json'])
+
+    # import images and convert to tensor
+    images = []
+    batch_img = []
+    annotations = []
+    batch_ann = []
+
+    # for each image
+    for i, iname, aname in zip(range(1, len(ann_names) + 1),
```

Comment:
Nothing wrong with this, but the enumerate function is quite handy for things like this:

`for i, iname, aname in enumerate(zip(image_names, ann_names), 1):`

Reply:
might be `for i, (iname, aname) in ...`